src/bindings/c/include/openvino/c/ov_node.h

```diff
@@ -35,70 +44,93 @@ typedef struct {
+OPENVINO_C_API(ov_status_e) ov_node_get_shape(ov_output_node_t* port, ov_shape_t* tensor_shape);
 
 /**
- * @brief Get the tensor name of ov_output_node list by index.
+ * @brief Get the tensor name of port list by index.
  * @ingroup node
- * @param nodes A pointer to the ov_output_node_list_t.
- * @param idx Index of the input tensor
+ * @param port_list A pointer to the ov_output_const_node_list_t.
+ * @param idx Index in the port_list.
  * @param tensor_name A pointer to the tensor name.
  * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
-ov_node_list_get_any_name_by_index(ov_output_node_list_t* nodes, size_t idx, char** tensor_name);
+ov_node_list_get_any_name_by_index(ov_output_const_node_list_t* port_list, size_t idx, char** tensor_name);
 
 /**
- * @brief Get the shape of ov_output_node.
+ * @brief Get the shape of port by index.
  * @ingroup node
- * @param nodes A pointer to the ov_output_node_list_t.
- * @param idx Index of the input tensor
- * @param tensor_shape tensor shape.
+ * @param port_list A pointer to the ov_output_const_node_list_t.
+ * @param idx Index in the port_list.
+ * @param shape The shape of the port.
  * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
-ov_node_list_get_shape_by_index(ov_output_node_list_t* nodes, size_t idx, ov_shape_t* shape);
+ov_node_list_get_shape_by_index(ov_output_const_node_list_t* port_list, size_t idx, ov_shape_t* shape);
 
 /**
- * @brief Get the partial shape of ov_output_node.
+ * @brief Get the partial shape of port with index in port_list.
  * @ingroup node
- * @param nodes A pointer to the ov_output_node_list_t.
- * @param idx Index of the input tensor
- * @param tensor_shape tensor shape.
+ * @param port_list A pointer to the ov_output_const_node_list_t.
+ * @param idx Index in the port_list.
+ * @param partial_shape Partial shape.
  * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
-ov_node_list_get_partial_shape_by_index(ov_output_node_list_t* nodes, size_t idx, ov_partial_shape_t** partial_shape);
+ov_node_list_get_partial_shape_by_index(ov_output_const_node_list_t* port_list, size_t idx, ov_partial_shape_t** partial_shape);
 
 /**
- * @brief Get the tensor type of ov_output_node.
+ * @brief Get the tensor type of port by index.
  * @ingroup node
- * @param nodes A pointer to the ov_output_node_list_t.
- * @param idx Index of the input tensor
+ * @param nodes A pointer to the ov_output_const_node_list_t.
+ * @param idx Index in the port_list.
  * @param tensor_type tensor type.
  * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
-ov_node_list_get_element_type_by_index(ov_output_node_list_t* nodes, size_t idx, ov_element_type_e* tensor_type);
+ov_node_list_get_element_type_by_index(ov_output_const_node_list_t* port_list, size_t idx, ov_element_type_e* tensor_type);
```

Comment:
Please take a look to get_shape proposal

Reply:
Update it in new commit